src/main/scala/exercises/07_Palindrome.scala

```diff
@@ -8,6 +8,11 @@ object Palindrome {
    *  Hint: the Scala collections API is your friend
    *  http://docs.scala-lang.org/overviews/collections/overview.html 
    */
-  def isPalindrome(s: String): Boolean = throw new NotImplementedError
+  def isPalindrome(s: String): Boolean = clean(s.toList) == clean(s.toList).reverse
 
+  private[this] def clean(s: List[Char]): List[Char] = s.toList match {
```

Comment:
This is an interesting way to clean the string of non-letters while simultaneously making all letters lowercase. I didn't think of doing it with case matching at all, since I'm more comfortable with functions than case matching. It's cool to see how case matching can be used to pick out the first character and recursively go through the rest. Although I'm not sure why `s.toList` was necessary since it is already a list of characters.